src/hypothesis/settings.py

```diff
@@ -198,6 +201,10 @@ def define_setting(cls, name, description, default, options=None):
         all_settings[name] = Setting(
             name, description.strip(), default, options)
         setattr(cls, name, SettingsProperty(name))
+        if cls.default:
+            setattr(cls.default, name, default)
+        for profile in cls._profiles.values():
```

Comment:
honestly im not 100% sure. You mentioned attribute lookup. When I register/load profiles I perform a copy so that may do it